<?xml version="1.0" encoding="UTF-8"?>

<launch>

  <arg name="param_file" default="$(find-pkg-share gnss_compass)/config/gnss_compass.yaml" />
  <!-- It is a choice whether to use gga or fix as input. -->
  <arg name="input_main_fix_topic" default="/main/mosaic/fix" description="Main gnss fix topic"/>
  <arg name="input_sub_fix_topic" default="/sub/mosaic/fix" description="Sub gnss fix topic"/>
  <arg name="input_main_gga_topic" default="/main/mosaic/gga" description="Main gnss gga topic"/>
  <arg name="input_sub_gga_topic" default="/sub/mosaic/gga" description="Sub gnss gga topic"/>
  <arg name="output_pose_topic" default="gnss_compass_pose" description="Estimated self position" />
  <arg name="output_pose_with_covariance_topic" default="gnss_compass_pose_with_covariance" description="Estimated self position" />

  <node pkg="gnss_compass" name="gnss_compass" exec="gnss_compass">
    <remap from="main_gnss_fix" to="$(var input_main_fix_topic)" />
    <remap from="sub_gnss_fix" to="$(var input_sub_fix_topic)" />
    <remap from="main_gnss_gga" to="$(var input_main_gga_topic)" />
    <remap from="sub_gnss_gga" to="$(var input_sub_gga_topic)" />
    <remap from="gnss_compass_pose" to="$(var output_pose_topic)" />
    <remap from="gnss_compass_pose_with_covariance" to="$(var output_pose_with_covariance_topic)" />
    <param from="$(var param_file)" />
  </node>

</launch>